### PR TITLE
Unversioned pages: overwrite home and developer docs on stable with dev version

### LINKED ISSUE #86: Unversioned (permanent) home page
## 📚 New content request
During today's Documentation working group meeting we have discussed using a permanent, unversioned home page to avoid versioning issues.

### Outline
Can be done together with napari/napari#4275

### PR DESCRIPTION
**Pull request by melissawm**
# Description
Creates pages that are always synced to latest by copying over index.html and the contents of the developers/ folder over their stable version.

There are maybe easier versions of this, happy to change strategy in that case 😄 

I've tested this successfully on my fork at melissawm/napari.github.io 

## Type of change

- [ ] Fixes or improves existing content
- [ ] Adds new content page(s)
- [x] Fixes or improves workflow, documentation build or deployment

# References
Fixes #86 

## Final checklist:
- [?] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added [alt text](https://webaim.org/techniques/alttext/) to new images included in this PR

### CODE REVIEW

File: .github/workflows/deploy_docs.yml [hunk 1]
```diff
@@ -72,3 +74,43 @@ jobs:
...
+
+    - name: Download artifact
+      uses: actions/download-artifact@master
+      with:
+        name: permanent_docs
+
+    - name: Copy permanent files to napari.github.io/stable
+      uses: dmnemec/copy_file_to_another_repo_action@main
+      env:
+        API_TOKEN_GITHUB: ${{ secrets.API_TOKEN_GITHUB }}
```
Comment (melissawm) [inline .github/workflows/deploy_docs.yml:109]: This token needs to be created on the napari.github.io repo

File: .github/workflows/deploy_docs.yml [hunk 2]
```diff
@@ -72,3 +74,43 @@ jobs:
...
+      with:
+        source_file: ${{ matrix.source_files }}
+        destination_repo: 'napari/napari.github.io'
+        destination_folder: 'stable'
+        destination_branch: 'gh-pages'
+        user_email: 'melissawm@gmail.com'
+        user_name: 'melissawm'
```
Comment (melissawm) [inline .github/workflows/deploy_docs.yml:116]: This needs to be replaced by the appropriate credentials

File: .github/workflows/deploy_docs.yml (hunk 1 above)
Comment (psobolewskiPhD) [reply .github/workflows/deploy_docs.yml:109]: I can make a secret within *this* repo (napari/docs) but not napari.github.io
Any chance the existing token used in the last deploy step: `deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}` could work?

**Comment by psobolewskiPhD** [comment]
Melissa walked me through this PR and there are some notes in the docs agenda:
https://hackmd.io/10DKrIMhREmk9sa_QRirlg?view#2023-08-16

It's quite clever and pretty safe: the unmodified stable and dev are both built and deployed, we're just adding a copy after, so if we wanted to abandon this strategy, it would just require commenting that out of the action and next merge to either repo would return the current status quo.

Unless someone has some other suggestion, I'd be in favor of merging and then we can figure out next steps?

File: .github/workflows/deploy_docs.yml (hunk 1 above)
Comment (melissawm) [reply .github/workflows/deploy_docs.yml:109]: we can try, I'm not sure! As long as it has the same repo permissions I think that should work?

**Comment by melissawm** [comment]
To clarify: the stable docs are not built every time - they were built once and now are in git history. So we could bring stable back by checking out the last commit before the overwrite. Still, it's there!

**Comment by melissawm** [comment]
After discussion, @Czaki suggested having two new jobs:
- One that builds the latest docs, and copies index.html and the developer documentation files over stable, as discussed previously;
- A second one that checks out the docs for the specific stable version (including those sources) and builds them too.
This is to prevent creating broken links and inconsistencies between versions.

Another tought was that we should have an extra check on docs PRs about broken links (either internal, using the sphinx linkchecker builder, or external, using something like the Lychee tool) to prevent merging PRs that break the docs.

Please let me know if I forgot anything cc @psobolewskiPhD @DragaDoncila

**Comment by melissawm** [comment]
I am closing this PR in favor of https://github.com/napari/napari.github.io/pull/393 - if we decide to reevaluate I can always reopen.

**Pull request closed without merging**